pkg/services/ngalert/state/state_tracker_test.go

```diff
@@ -27,12 +37,24 @@ func TestProcessEvalResults(t *testing.T) {
 					Instance: data.Labels{"label1": "value1", "label2": "value2"},
 				},
 			},
-			expectedCacheEntries: 1,
-			expectedState:        eval.Normal,
-			expectedResultCount:  0,
+			expectedState:              eval.Normal,
+			expectedReturnedStateCount: 0,
+			expectedResultCount:        1,
+			expectedCacheEntries: []AlertState{
+				{
+					UID:         "test_uid",
+					CacheId:     "test_uid label1=value1, label2=value2",
+					Labels:      data.Labels{"label1": "value1", "label2": "value2"},
+					State:       eval.Normal,
+					Results:     []eval.State{eval.Normal},
+					StartsAt:    strfmt.DateTime{},
+					EndsAt:      strfmt.DateTime{},
+					EvaluatedAt: strfmt.DateTime(evaluationTime),
+				},
+			},
 		},
 		{
-			desc: "given a state change from normal to alerting",
+			desc: "given a state change from normal to alerting for a single entity",
```

Comment:
I think that this is not realistic scenario because it violates [this](https://github.com/grafana/grafana/blob/ce679a67b758cb081df1ec1c14cf6f8c36ee7655/pkg/services/ngalert/database_mig.go#L98) precondition (each evaluation result of a specific alert is uniquely identified from its labels):
https://github.com/grafana/grafana/blob/4a6ad4ad88d856dc6136872ac2d3f4443c399166/pkg/services/ngalert/state/state_tracker_test.go#L59-L68

Reply:
It's definitely contrived for the purposes of demonstrating that the logic to transition from `normal` to `alerting`. I'm attempting to simulate subsequent evaluations without having to spin up an entire instance of the scheduler.